fcrepo4/fcrepo4
# Ensure that <> in RDF bodies uses 'this' resource URI

Resolves: https://jira.lyrasis.org/browse/FCREPO-3259
Incremental resolution for: https://jira.lyrasis.org/browse/FCREPO-3223

# What does this Pull Request do?
Establishes resource URI in FedoraLDP for all creation requests
Fixes more FedoraLdpIT tests

# How should this be tested?
1. PUT and POST of containers and binaries should work as before
1. PUT and POST of containers with RDF bodies should create the correct RDF subjects
1. POST with a slug of an existing resource should work as before

The following `curl` examples can be helpful in manually testing this pull-request
```
# Create and inspect empty PUT resource
curl -XPUT localhost:8080/rest/put0
curl  localhost:8080/rest/put0

# Create and inspect empty POST resource
curl -XPOST localhost:8080/rest/ -H"slug: post0"
curl  localhost:8080/rest/post0

# Create and inspect empty POST resource, with compound slug
curl -XPOST localhost:8080/rest/ -H"slug: post0/1"
curl  localhost:8080/rest/post0/1

# Create and inspect PUT non-RDF resource with body
# ..where binary.txt is:
# "Hello World"
curl -i -XPUT --data-binary @binary.txt -H"Content-Type: text/plain"  localhost:8080/rest/put1
curl  localhost:8080/rest/put1
curl  localhost:8080/rest/put1/fcr:metadata

# Create and inspect POST non-RDF resource with body
# ..where binary.txt is:
# "Hello World"
curl -i -XPOST --data-binary @binary.txt -H"Content-Type: text/plain"  localhost:8080/rest/ -H"slug: post1"
curl  localhost:8080/rest/post1/fcr:metadata
curl  localhost:8080/rest/post1

# re-POST with an existing slug
curl -i -XPOST --data-binary @binary.txt -H"Content-Type: text/plain"  localhost:8080/rest/ -H"slug: post1"
curl http://localhost:8080/rest/c49a58b1-d71f-4fe5-8501-b92a6f65f98a
curl http://localhost:8080/rest/c49a58b1-d71f-4fe5-8501-b92a6f65f98a/fcr:metadata

# Create and inspect PUT RDF resource with body
# ..where body.ttl is:
# "<> <http://purl.org/dc/elements/1.1/title> "Some title""
curl -XPUT --data-binary @body.ttl -H"Content-Type: text/turtle" http://localhost:8080/rest/put2
curl http://localhost:8080/rest/put2

# Create and inspect POST RDF resource with body
# ..where body.ttl is:
# "<> <http://purl.org/dc/elements/1.1/title> "Some title""
curl -XPOST --data-binary @body.ttl -H"Content-Type: text/turtle" http://localhost:8080/rest/ -H"slug: post2"
curl http://localhost:8080/rest/post2

# re-POST with an existing slug
curl -XPOST --data-binary @body.ttl -H"Content-Type: text/turtle" http://localhost:8080/rest/ -H"slug: post2"
```

# Interested parties
@whikloj and @fcrepo4/committers
